llvm/include/llvm/Demangle/ItaniumDemangle.h

```diff
@@ -5737,6 +5737,7 @@ Node *AbstractManglingParser<Derived, Alloc>::parseTemplateParamDecl(
   }
 
   if (consumeIf("Tk")) {
+    ScopedOverride<bool> SaveInConstraintExpr(InConstraintExpr, true);
```

Comment:
Since this is a temporary fix, it would be helpful to add comments that explain both what the newly-added line of code is doing and why it’s necessary for now.

Reply:
Sure, there's a comment in https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/Demangle/ItaniumDemangle.h#L5679, but I can also add a similar one here.